src/nevermined/api/AssetsApi.ts

```diff
@@ -0,0 +1,464 @@
+        const { attributes } = ddo.findServiceByType('metadata')
+        const { files } = attributes.main
+
+        const { serviceEndpoint, index } = ddo.findServiceByType(serviceType)
+
+        if (!serviceEndpoint) {
+            throw new AssetError(
+                'Consume asset failed, service definition is missing the `serviceEndpoint`.'
+            )
+        }
+
+        this.logger.log('Consuming files')
+
+        resultPath = resultPath
+            ? `${resultPath}/datafile.${ddo.shortId()}.${index}/`
+            : undefined
+
+        const accessToken = await this.nevermined.utils.jwt.getDownloadGrantToken(
+            ddo.id,
+            ownerAccount
+        )
+        const headers = {
+            Authorization: 'Bearer ' + accessToken
+        }
+        return this.nevermined.services.node.downloadService(
+            files,
+            resultPath,
+            fileIndex,
+            isToDownload,
+            headers
+        )
+    }
+
+
+    /**
+     * It grants permissions to an account for a specific asset represented by a DID.
+     * Only can be called by the asset owner.
+     * @param did - The unique identifier of the assert
+     * @param address - The account to grant the permissions
+     * @param account - Account sending the request. It must be the owner of the asset
+     * @param params  - Transaction parameters
+     */    
+    public async grantPermissions(
+        did: string,
+        address: string,
+        account: Account,
+        params?: TxParameters
+    ) {
+        return await this.nevermined.keeper.didRegistry.grantPermission(
+            did,
+            address,
+            account.getId(),
+            params
+        )
+    }
+
+    /**
+     * It revokes permissions to an account for a specific asset represented by a DID.
+     * Only can be called by the asset owner.
+     * @param did - The unique identifier of the assert
+     * @param address - The account to revoke the permissions
+     * @param account - Account sending the request. It must be the owner of the asset
+     * @param params  - Transaction parameters
+     */
+    public async revokePermissions(
+        did: string,
+        address: string,
+        account: Account,
+        params?: TxParameters
+    ) {
+        return await this.nevermined.keeper.didRegistry.revokePermission(
+            did,
+            address,
+            account.getId(),
+            params
+        )
+    }
+
+    /**
+     * Checks if an account with a specific address has permissions to a specific asset represented by a DID
+     * @param did - The unique identifier of the asset to check the permissions
+     * @param address - The address of the account to check the permissions
+     * @returns True if the address has permissions on the asset
+     */
+    public async getPermissions(did: string, address: string) {
+        return await this.nevermined.keeper.didRegistry.getPermission(did, address)
+    }
```

Comment:
this doesn't seem to be used anywhere

Reply:
this methods are needed to manage the providers associated to an asset